pyverbs/qp.pyx

```diff
@@ -1260,9 +1260,10 @@ cdef class QPEx(QP):
         :return: An initialized QPEx object
         """
         super().__init__(creator, init_attr, qp_attr)
-        self.qp_ex = v.ibv_qp_to_qp_ex(self.qp)
-        if self.qp_ex == NULL:
-            raise PyverbsRDMAErrno('Failed to create extended QP')
+        if init_attr.comp_mask & v.IBV_QP_INIT_ATTR_SEND_OPS_FLAGS:
+            self.qp_ex = v.ibv_qp_to_qp_ex(self.qp)
+            if self.qp_ex == NULL:
+                raise PyverbsRDMAErrno('Failed to create extended QP')
```

Comment:
@EdwardSro Could you please tell what will happen if someone tries to create a QPEx object without setting IBV_QP_INIT_ATTR_SEND_OPS_FLAGS bit in the comp_mask?

Reply:
I think I should revert that in V2.
You can take a look at the QP object. ibv_create_qp takes a PD while ibv_create_qp_ex takes a context instead, but both return an ibv_qp.
QP class has a creator parameter, which can be either PD or Context instance, upon that we decide if we are creating a qp or a qp_ex. Since both of them return ibv_qp in C, we also have the same QP class in python.

Back to your question, he'll have a qp (which actually is an extended qp) but he won't have access to the new send APIs, if he tries to do so he can fail on segfault since self.qp_ex is not defined...

Reply:
Eventually, I could not remove this check, since each provider-specific that wants to inherit from QPEX but doesn't always need the new post send API, would fail otherwise. I added a debug print in case the user doesn't pass IBV_QP_INIT_ATTR_SEND_OPS_FLAGS.